content/articles/building-a-simple-translation-app-using-python-for-beginners/index.md

```diff
@@ -0,0 +1,168 @@
+
+After selecting those options, this is the command you're going to run to install PyTorch:
+
+```python
+pip3 install torch==1.8.1+cpu torchvision==0.9.1+cpu torchaudio==0.8.1 -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html
+```
+Let's now go ahead and install our other dependencies, the Hugging Face Transformers and Gradio. 
+```python
+pip install transformers ipywidgets gradio --upgrade
+```
+> Please note that this command can be used in the command-line interface for a local machine and can also be run as a cell in the Colab notebook. You only add an exclamation `!` mark before the `pip` command when installing on the Colab. This way: `!pip install transformers ipywidgets gradio --upgrade`
+
+Our dependencies are now installed successfully!
+
+`transformers` is going to give you our translation pipeline.
+`ipywidgets` provides you with the progress bar as you're downloading the model.
+`gradio` gives you a nice way to demonstrate and interact with your ML model.
+
+Now that we've installed our dependencies, the next step involves importing them into our model.
+
+To import them, let's issue the following commands:
+
+```python
+import gradio as gr
+from transformers import pipeline
+```
+Now that that is done, let's go ahead and download the pre-trained translation model.
+
+### Downloading the pre-trained translation model
+Hugging Face has many transformer pipelines. You can check them on this [documentation](https://huggingface.co/transformers/main_classes/pipelines.html). For this tutorial, you will select the `TranslationPipeline`, which is a task-specific pipeline that allows translation from one language to another. In the documentation, they state that you can use the `pipeline()` method and the task identifier `translation_xx_to_yy` to translate, `xx` being the language you want to translate from and `yy` the language you want to translate to. For this example, we are building a simple translation app that can translate from English to French. 
+
+To do this, let's type the following code:
+
+```python
+translation_pipeline = pipeline ('translation_en_to_fr')
+```
+Let's try it out. So, for example, if I wanted to translate the sentence, "I can't wait to watch Money Heist", we can do it like this:
+
+```python
+translation_pipeline ('I cannot wait to watch Money Heist')
+```
+This will be the output:
+
+```python
+
+[{'translation_text': 'Je ne peux attendre de regarder Money Heist'}]
+
+```
+
+Amazing right? The translation pipeline is translating our text from English to French. 
+
+Lastly, we are going to create a function to wrap up all these and display it on a beautiful user interface. We'll be leveraging Gradio for this task.
+
+### Building the web app for interaction using Gradio
+This web app allows a user to pass through some text and have it translated.
```

Comment:
Do add a few sentences about what a web-app is and link an external resource or two to it.